sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -925,6 +925,70 @@ export class ManagementClient extends LinkEntity<RequestResponseLink> {
     }
   }
 
+  /**
+   * Batch-delete messages.
+   *
+   * @param messageCount - number of messages to delete in a batch.
+   * @param enqueueTimeUtcOlderThan - Delete messages whose enqueue time (UTC) are older than this.
+   * @returns number of messages deleted.
+   */
+  async batchDeleteMessages(
+    messageCount: number,
+    enqueueTimeUtcOlderThan?: Date,
+    sessionId?: string,
+    options: OperationOptionsBase & SendManagementRequestOptions = {},
+  ): Promise<number> {
+    throwTypeErrorIfParameterMissing(this._context.connectionId, "messageCount", messageCount);
+    throwTypeErrorIfParameterTypeMismatch(
+      this._context.connectionId,
+      "messageCount",
+      messageCount,
+      "number",
+    );
+
+    if (isNaN(messageCount) || messageCount < 1) {
+      throw new TypeError("'messageCount' must be a number greater than 0.");
+    }
+
+    try {
+      const messageBody: any = {};
+      messageBody[Constants.messageCount] = types.wrap_int(messageCount);
+      messageBody[Constants.enqueuedTimeUtc] = enqueueTimeUtcOlderThan ?? new Date();
+      if (isDefined(sessionId)) {
+        messageBody[Constants.sessionIdMapKey] = sessionId;
+      }
+
+      const updatedOptions = await this.initWithUniqueReplyTo(options);
+      const request: RheaMessage = {
+        body: messageBody,
+        reply_to: this.replyTo,
+        application_properties: {
+          operation: Constants.operations.batchDeleteMessages,
+        },
+      };
+      if (updatedOptions?.associatedLinkName) {
+        request.application_properties![Constants.associatedLinkName] =
+          updatedOptions?.associatedLinkName;
+      }
+      request.application_properties![Constants.trackingId] = generate_uuid();
+      receiverLogger.verbose("%s batch delete request body: %O.", this.logPrefix, request.body);
+
+      const result = await this._makeManagementRequest(request, receiverLogger, updatedOptions);
+      if (result.application_properties!.statusCode === 200) {
```

Comment:
does `_makeManagementRequest` throw an error if the response status code is not 200? If not, should we throw here?

Reply:
Updated to return 0 for 204 and throw errors for others.